colorizer_data/utils.py

```diff
@@ -129,34 +138,109 @@ def remap_segmented_image(
     return (seg_remapped, lut)
 
 
-def update_collection(collection_filepath, dataset_name, dataset_path):
+def update_metadata(
+    metadata: Optional[Union[CollectionMetadata, ColorizerMetadata]],
+    *,
+    default_name: Optional[str] = None,
+):
+    """
+    Updates the following fields in a dataset or collection metadata object:
+    - date_created: Sets to current time if none exists.
+    - last_modified: Sets to current time.
+    - _revision: Sets to 0 if none exists; otherwise increments by 1.
+    - _writer_version: Sets to `types.CURRENT_VERSION`.
+    - name: Sets name if none exists and a `default_name` argument was provided.
+
+    Args:
+        metadata (CollectionMetadata | ColorizerMetadata): The metadata object to update.
+        default_name (str): The name of the collection or dataset to use if the metadata has none.
+    """
+    current_time = datetime.now(timezone.utc).strftime(DATETIME_FORMAT)
+
+    # Update creation date if missing
+    if metadata.date_created is None:
+        metadata.date_created = current_time
+    # Update revision number
+    revision = metadata._revision
+    if revision is None:
+        metadata._revision = 0
+    else:
+        metadata._revision = revision + 1
+    # Update data version + modified timestamp
+    metadata.last_modified = current_time
+    metadata._writer_version = CURRENT_VERSION
+
+    # Use default dataset name from writer constructor if no name was loaded
+    # (will be overridden by `metadata.name` argument if provided)
+    if metadata.name is None and default_name is not None:
+        metadata.name = default_name
```

Comment:
because this is the only part that depends on a parameter (setting the name),  I would consider pulling this out of this function, so that update_metadata only does updating of fields it can do automatically.  Then I'd rename the function to bump_versions_and_timestamps